genrl/deep/agents/ppo1/ppo_temp.py

```diff
@@ -0,0 +1,166 @@
+        self.entropy_coeff = entropy_coeff
+        self.value_coeff = value_coeff
+
+        self.create_model()
+
+    def create_model(self):
+        # Instantiate networks and optimizers
+        state_dim, action_dim, disc, action_lim = get_env_properties(self.env)
+        self.policy_new = get_model("p", self.network_type)(
+            state_dim, action_dim, self.layers, disc=disc, action_lim=action_lim
+        )
+        self.policy_new = self.policy_new.to(self.device)
+
+        self.value_fn = get_model("v", self.network_type)(state_dim, action_dim).to(
+            self.device
+        )
+
+        # load paramaters if already trained
+        if self.load_model is not None:
+            self.load(self)
+            self.policy_new.load_state_dict(self.checkpoint["policy_weights"])
+            self.value_fn.load_state_dict(self.checkpoint["value_weights"])
+            for key, item in self.checkpoint.items():
+                if key not in ["policy_weights", "value_weights", "save_model"]:
+                    setattr(self, key, item)
+            print("Loaded pretrained model")
+
+        self.optimizer_policy = opt.Adam(
+            self.policy_new.parameters(), lr=self.lr_policy
+        )
+        self.optimizer_value = opt.Adam(self.value_fn.parameters(), lr=self.lr_value)
+
+        self.rollout = RolloutBuffer(
+            self.rollout_size,
+            self.env.observation_space,
+            self.env.action_space,
+            n_envs=self.env.n_envs,
+        )
+
+    def select_action(self, state: np.ndarray) -> np.ndarray:
+        state = torch.as_tensor(state).float().to(self.device)
+        # create distribution based on policy output
+        action, c_new = self.policy_new.get_action(state, deterministic=False)
+        value = self.value_fn.get_value(state)
+
+        return action.detach().cpu().numpy(), value, c_new.log_prob(action)
+
+    def evaluate_actions(self, obs, old_actions):
+        value = self.value_fn.get_value(obs)
+        _, dist = self.policy_new.get_action(obs)
+        return value, dist.log_prob(old_actions), dist.entropy()
+
+    def get_traj_loss(self, values, dones):
+        self.rollout.compute_returns_and_advantage(
+            values.detach().cpu().numpy(), dones, use_gae=True
+        )
+
+    def update_policy(self):
+
+        for rollout in self.rollout.get(256):
+            actions = rollout.actions
+
+            if isinstance(self.env.action_space, gym.spaces.Discrete):
+                actions = actions.long().flatten()
+
+            values, log_prob, entropy = self.evaluate_actions(
+                rollout.observations, actions
+            )
+
+            advantages = rollout.advantages
+            advantages = (advantages - advantages.mean()) / (advantages.std() + 1e-8)
+
+            ratio = torch.exp(log_prob - rollout.old_log_prob)
+
+            policy_loss_1 = advantages * ratio
+            policy_loss_2 = advantages * torch.clamp(ratio, 1 - 0.2, 1 + 0.2)
```

Comment:
Clip param should be used here, instead of 0.2

Reply:
Ok will make this change.